### PR TITLE
Replaced custom config package with viper

### PR DESCRIPTION
**Pull request by dszlachta**
- Added Viper
- Set defaults for configuration values
- Enabled reading environment variables (e.g. `[dev] ipfs_gateway` -> `DEV_IPFS_GATEWAY=mygateway.example.com`)
- Loading config files into structs to make them first-class citizens in our code

### CODE REVIEW

File: src/apps/chifra/pkg/config/block_scrape.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+// ReadBlockScrape reads the configuration located in blockScrape.toml file
+func ReadBlockScrape() *BlockScrape {
+	if !blockScrapeRead {
+		MustReadConfig(blockScrapeViper, &cachedBlockScrape)
+
+		// Validate the URL to ensure we have it in the correct format, so that ethClient.Dial
+		// will not panic
+		if !strings.HasPrefix(cachedBlockScrape.Dev.IpfsGateway, "http") {
```
Comment (tjayrush) [inline src/apps/chifra/pkg/config/block_scrape.go:69]: I usually also check for an ending `/' whenever I read from a config file. If it's not there, I add it. In this way I always know that any config file path or URL only needs the subfolder or route to be added and I don't have to check for the trailing 'slash'.

File: src/apps/chifra/pkg/config/block_scrape.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+type BlockScrape struct {
+	Requires       blockScrapeRequires
+	Dev            blockScrapeDev
+	UnchainedIndex blockScrapeUnchainedIndex
+}
+
+// init sets up default values for the given configuration
+func init() {
```
Comment (tjayrush) [inline src/apps/chifra/pkg/config/block_scrape.go:51]: So, just for my information. Is it true that a `init` function in any file will get called prior to startup? In other words, prior to `main`, so we can count on these values being here before main?

File: src/apps/chifra/pkg/config/global_config.go [hunk 3]
```diff
@@ -12,7 +12,14 @@
  *-------------------------------------------------------------------------------------------*/
```
Comment (tjayrush) [inline src/apps/chifra/pkg/config/global_config.go]: Can we please rename this file `trueblocks.go`? In that way, when I look in this folder, each file will be named the same as it's subcommand route except for this one. And the names of these go files will reflect the actual names of the toml files they read.

File: src/apps/chifra/pkg/config/utils.go [hunk 4]
```diff
@@ -64,29 +65,20 @@ func GetConfigPath(fileName string) string {
...
-		fileNameWithExt += ".toml"
-	}
-
-	filePath := GetConfigPath(fileNameWithExt)
+// MustReadConfig calls v's ReadInConfig and fills values in the
+// given targetStruct. Any error will result in a call to logger.Fatal
+func MustReadConfig(v *viper.Viper, targetStruct interface{}) {
+	v.AddConfigPath(GetConfigPath(""))
+	v.AutomaticEnv()
+	v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
```
Comment (tjayrush) [inline src/apps/chifra/pkg/config/utils.go:74]: See the viper usages in `./src/go-apps/blaze/cmd/options.go` (I think). We prepend `TB_` in front of the environment variables. In this way, they are all easily discoverable with `env | grep TB_` from the command line. But don't change the settings in the files, just the environment variables.

**Comment by tjayrush** [comment]
This looks great. Just close these few issues and we can merge it.

File: src/apps/chifra/pkg/config/block_scrape.go (hunk 1 above)
Comment (dszlachta) [reply src/apps/chifra/pkg/config/block_scrape.go:69]: This is exactly why the code later uses `path.Join`, which will add ending `/` if it's missing

File: src/apps/chifra/pkg/config/block_scrape.go (hunk 2 above)
Comment (dszlachta) [reply src/apps/chifra/pkg/config/block_scrape.go:51]: Yes, it's described here (5th paragraph from the end of the section): https://golang.org/ref/spec#Package_initialization

File: src/apps/chifra/pkg/config/global_config.go (hunk 3 above)
Comment (dszlachta) [reply src/apps/chifra/pkg/config/global_config.go]: Definitely, good idea

**Review by tjayrush — COMMENTED** [review]
I forgot to submit this one comment...

**Pull request merged**